vispy/visuals/line/arrow.py

```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+"""
+Arrows are a subclass of line visuals, which adds the ability to put several
+heads on a line.
+"""
+
+from __future__ import division
+
+import numpy as np
+
+from ... import glsl, gloo
+from ..visual import Visual, CompoundVisual
+from .line import LineVisual
+
+
+ARROW_TYPES = (
+    'stealth',
+    'curved',
+    'angle_30',
+    'angle_60',
+    'angle_90',
+    'triangle_30',
+    'triangle_60',
+    'triangle_90',
+    'inhibitor_round'
+)
+
+
+class _ArrowHeadVisual(Visual):
+    """
+    ArrowHeadVisual: several shapes to put on the end of a line.
+    This visual differs from MarkersVisual in the sense that this visual
+    calculates the orientation of the visual on the GPU, by calculating the
+    tangent of the line between two given vertices.
+
+    This is not really a visual you would use on your own,
+    use :class:`ArrowVisual` instead.
+
+    Parameters
+    ----------
+    parent : ArrowVisual
+        This actual ArrowVisual this arrow head is part of.
+    """
+
+    ARROWHEAD_VERTEX_SHADER = glsl.get('arrowheads/arrowheads.vert')
+    ARROWHEAD_FRAGMENT_SHADER = glsl.get('arrowheads/arrowheads.frag')
+
+    _arrow_vtype = np.dtype([
+        ('v1', 'f4', 2),
+        ('v2', 'f4', 2),
+        ('size', 'f4', 1),
+        ('color', 'f4', 4),
+        ('linewidth', 'f4', 1)
+    ])
+
+    def __init__(self, parent):
+
+        Visual.__init__(self, self.ARROWHEAD_VERTEX_SHADER,
+                        self.ARROWHEAD_FRAGMENT_SHADER)
+
+        self._parent = parent
+        self._arrow_vbo = gloo.VertexBuffer()
+        self._draw_mode = 'points'
+
+    def _prepare_transforms(self, view):
+        xform = view.transforms.get_transform()
+        view.view_program.vert['transform'] = xform
+
+    def _prepare_draw(self, view=None):
+
+        if self._parent._arrows_changed:
+            v = self._prepare_vertex_data()
+            self._arrow_vbo.set_data(v)
+
+        self.shared_program.bind(self._arrow_vbo)
+
+        self.shared_program['antialias'] = 1.0
+        self.shared_program.frag['arrow_type'] = self._parent.arrow_type
+        self.shared_program.frag['fill_type'] = "filled"
+
+    def _prepare_vertex_data(self):
+        arrows = self._parent.arrows
+
+        num_arrows = arrows.shape[0]
+        v = np.zeros(num_arrows, dtype=self._arrow_vtype)
+
+        v['v1'] = arrows[:, 0:2]
+        v['v2'] = arrows[:, 2:4]
+
+        v['size'][:] = self._parent.arrow_size
+        v['color'][:] = self._parent.line_visual._interpret_color()
+        v['linewidth'][:] = self._parent.line_visual.width
+
+        return v
+
+
+class ArrowVisual(CompoundVisual):
```

Comment:
Hmm, we *could* leave it as a superclass of `LineVisual` since that's already a `CompoundVisual` as @campagnola suggests. It seems cleaner / more explict to me to be a superclass of `CompoundVisual` because then it's clear what visuals are actually being subclassed. But I think the code duplication necessary e.g. to duplicate all the parameters is not worth the DRY cost, so it might be worth reverting this change. @sh4wn WDYT?